repairnator/src/main/java/fr/inria/spirals/repairnator/process/ProjectScanner.java

```diff
@@ -254,23 +235,39 @@ private boolean testBuild(Build build, boolean targetFailing) {
         return false;
     }
 
-    private List<Build> getListOfBuildsFromRepo(List<Repository> repos, boolean targetFailing) {
-        List<Build> result = new ArrayList<Build>();
+    private List<BuildToBeInspected> getListOfBuildsFromRepo(List<Repository> repos, RepairMode mode, boolean targetFailing) {
+        List<BuildToBeInspected> buildsToBeInspected = new ArrayList<BuildToBeInspected>();
 
-        this.buildsId = new ArrayList<Integer>();
         for (Repository repo : repos) {
             List<Build> repoBuilds = BuildHelper.getBuildsFromRepositoryWithLimitDate(repo, this.limitDate);
 
             for (Build build : repoBuilds) {
                 this.totalScannedBuilds++;
                 if (testBuild(build, targetFailing)) {
-                    result.add(build);
-                    this.buildsId.add(build.getId());
+                    if (mode == RepairMode.FORBEARS) {
```

Comment:
Because for RepairNator it is not necessary to do things with the previous build.